javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1449,7 +1449,7 @@ public static Object InferType(String attrValue) {
         }
 
         try {
-            Double ret = new Double(Double.parseDouble(attrValue));
+            Double ret = MathUtils.parseDoubleSafe(attrValue);

```

Comment:
are there any performance implications of this? just wondering if we should only use our custom thing conditionally on android version or something.